basyx.submodelrepository-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/http/TestSubmodelRepositoryHTTP.java

```diff
@@ -87,13 +89,48 @@ public void getSpecificSubmodel() throws ParseException, IOException {
 
 	@Test
 	public void getSpecificSubmodelNonExisting() throws IOException {
-		CloseableHttpResponse response = executeGetOnURL(submodelAccessURL + "/nonExisting/submodel");
+		CloseableHttpResponse response = getSubmodel("nonExisting");
 
 		assertEquals(HttpStatus.NOT_FOUND.value(), response.getCode());
 	}
 
+
+	@Test
+	public void updateExistingSubmodel() throws IOException, ParseException {
+		String id = "7A7104BDAB57E184";
```

Comment:
To ensure that the id of `submodelJSON `is the same id as the one used in `expectedSubmodelJSON`, it would be appropriate to extract that id from `expectedSubmodelJSON `and use it e.g. like this `putSubmodel("extractedId", expectedSubmodelJSON)`. This ensures that the file `SingleSubmodelUpdate.json` could be changed without necessarily touching the Test.